### DISCUSSION TITLE
PIP-29: Batch Transactions

### DISCUSSION THREAD

**Discussion by maxipaz** [pactus-project/PIPs#168] [PIPs]
[PIP-29: Batch Transactions](https://pips.pactus.org/PIPs/pip-29)

**Comment by b00f** [comment]
Thanks @maxipaz for drafting the "Batch Transaction" PIP.

The definition of Batch Transaction is now clear, and it requires more discussion and thought. 
Based on the PIP, we can define the Batch Transaction in these ways:

The first approach is to define the Batch Transaction as a transaction with **multiple payloads**.

This approach is a general way to define the batch transaction and probably has a simpler implementation. However, it has two disadvantages:
- For transactions with similar payloads, like two transfers, the size of transactions gets bigger since we repeat the sender address.
- Constructing these transactions using CLI is complicated.

The second approach is to define the Batch Transaction as a transaction with **multiple receivers**.

This approach is not general and is case-based. 
It can apply only to Transfer, Bond, and Withdraw transactions. 
However, it has some advantages:

- Reducing the size of transactions.
- It is easy to construct transactions using CLI. For example, we can create a batch transaction like:

```
./pactus-wallet tx transfer <FROM> <TO_1> <AMOUNT_1> --extra-receivers "<TO_2>:<AMOUNT_2>"
```

**Reply by maxipaz** [reply]
Just to share a thought I had while writing this: IMHO the biggest benefit about Batch Transactions is the transaction fee save... But if we aim to have gas-less transactions, this benefit won't be important.

**Reply by b00f** [reply]
It can still save fees. In a gas-less model, transactions are not always free. For each address, 2-3 transactions per day might be free.
